File: app/src/main/java/com/wadektech/eventshub/ui/ProfessionaEventsActivity.java
package com.wadektech.eventshub.ui;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import com.wadektech.eventshub.R;
import com.wadektech.eventshub.adapter.ProfessionalEventsAdapter;
import com.wadektech.eventshub.models.ProfessionalEvents;
import com.wadektech.eventshub.viewmodels.ProfessionalEventsViewModel;

import java.util.ArrayList;
import java.util.List;

public class ProfessionaEventsActivity extends AppCompatActivity implements ProfessionalEventsAdapter.OnSingleProfEventClicked {
    RecyclerView mRecycler ;
    LinearLayoutManager mLayout ;
    ProfessionalEventsAdapter mProfAdapter ;
    List<ProfessionalEvents> professionalEventsList ;
    ProfessionalEventsViewModel professionalEventsViewModel ;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_professiona_events);

        mRecycler = findViewById(R.id.rec_prof_events);
        mLayout = new LinearLayoutManager(this) ;
        mRecycler.setHasFixedSize(true);
        mRecycler.setLayoutManager(mLayout);

        professionalEventsList = new ArrayList<>() ;

        professionalEventsViewModel = ViewModelProviders.of(this).get(ProfessionalEventsViewModel.class);
        professionalEventsViewModel.getAllProfessionalEvents().observe(this, new Observer<List<ProfessionalEvents>>() {
            @Override
            public void onChanged(List<ProfessionalEvents> professionalEventsList) {
                mProfAdapter = new ProfessionalEventsAdapter(professionalEventsList, ProfessionaEventsActivity.this, ProfessionaEventsActivity.this::singleProfessionalEventClicked);
                mRecycler.setAdapter(mProfAdapter);
            }
        });
    }

    @Override
    public void singleProfessionalEventClicked(int position) {
        Intent intent = new Intent(ProfessionaEventsActivity.this, MoreInfo.class);
        startActivity(intent);
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        Intent intent = new Intent(ProfessionaEventsActivity.this, MainActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/wadektech/eventshub/ui/ProfessionaEventsActivity.kt
package com.wadektech.eventshub.ui

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.wadektech.eventshub.R
import com.wadektech.eventshub.adapter.ProfessionalEventsAdapter
import com.wadektech.eventshub.adapter.ProfessionalEventsAdapter.OnSingleProfEventClicked
import com.wadektech.eventshub.database.MainEventsRoomDatabase
import com.wadektech.eventshub.models.ProfessionalEvents
import com.wadektech.eventshub.repository.EventsHubRepository
import com.wadektech.eventshub.utils.EventsHubViewModelFactory
import com.wadektech.eventshub.viewmodels.EventsHubViewModel
import java.util.*

class ProfessionaEventsActivity : AppCompatActivity(), OnSingleProfEventClicked {
    private lateinit var mRecycler: RecyclerView
    private lateinit var mLayout: LinearLayoutManager
    private lateinit var mProfAdapter: ProfessionalEventsAdapter
    private lateinit var eventsHubViewModel: EventsHubViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_professiona_events)

        mRecycler = findViewById(R.id.rec_prof_events)

        initRecyclerview()

        mProfAdapter = ProfessionalEventsAdapter(this)
        mRecycler.adapter = mProfAdapter

        val db = MainEventsRoomDatabase(this)
        val repo = EventsHubRepository(db)
        val factory = EventsHubViewModelFactory(repo)

        eventsHubViewModel = ViewModelProvider(this, factory).get(EventsHubViewModel::class.java)
        eventsHubViewModel.getAllProfEvents.observe(this, Observer {
            mProfAdapter.submitList(it)
        })
    }

    override fun singleProfessionalEventClicked(position: Int) {
        val intent = Intent(this@ProfessionaEventsActivity, MoreInfo::class.java)
        startActivity(intent)
    }

    override fun onBackPressed() {
        super.onBackPressed()
        val intent = Intent(this@ProfessionaEventsActivity, MainActivity::class.java)
        startActivity(intent)
    }

    private fun initRecyclerview(){
        mRecycler.setHasFixedSize(true)
        mLayout = LinearLayoutManager(this)
        mRecycler.layoutManager = mLayout
    }
}